core/src/main/scala/caliban/schema/ArgBuilder.scala

```diff
@@ -205,6 +209,16 @@ trait ArgBuilderInstances extends ArgBuilderDerivation {
 
   implicit lazy val upload: ArgBuilder[Upload] = {
     case StringValue(v) => Right(Upload(v))
-    case other          => Left(ExecutionError(s"Can't build an Upload from $other"))
+    case other          => Left(InvalidInputArgument("Upload", other.toString))
   }
 }
+
+case object InvalidInputArgument extends NoStackTrace {
+  override def getMessage: String = "invalid input argument"
+
+  private[caliban] def apply(expected: String, actual: String): ExecutionError =
```

Comment:
`expected` and `actual` are a bit misleading because it sounds like they should be equal like in tests.
How about `argType` and `argValue`?

We could potentially accept `Any` for `ArgValue` to avoid doing `toString` multiple times too.

Reply:
Yeah I think that's better 👍